bundles/SeoBundle/src/Redirect/RedirectHandler.php

```diff
@@ -135,7 +135,7 @@ private function matchRegexRedirect(
 
         // check for a site
         if ($redirect->getSourceSite() || $sourceSite) {
-            if (!$sourceSite || $sourceSite->getId() !== $redirect->getSourceSite()) {
+            if (!$sourceSite || ($redirect->getSourceSite() && $sourceSite->getId() !== $redirect->getSourceSite())) {
```

Comment:
a little OT but noticed that the conditions of these 2 nested IF blocks need a little refactoring, the conditions are too crowded/repeated, maybe can achieve the same with some null coalescing or 2 distinct IF

```
        if ($redirect->getSourceSite() || $sourceSite) {
            if (!$sourceSite || ($redirect->getSourceSite() && $sourceSite->getId() !== $redirect->getSourceSite())) {
```

Reply:
Right. Have refactored it to
```php
// check for a site
if ($sourceSite && $sourceSite->getId() !== $redirect->getSourceSite()) {
  return null;
}
```

Reply:
mhhh, there's something missing, if the former logic is right
````
if ($redirect->getSourceSite() || $sourceSite) {
    if (!$sourceSite || ($redirect->getSourceSite() && $sourceSite->getId() !== $redirect->getSourceSite())) {
        return null;
    }
}
````
given `A` is `$redirect->getSourceSite()` and `B` is `$sourceSite`

A) if A is truthy but B is falsy, then return null due to first condition in nested IF
B) if A is truthy and B is truthy, then check B->getId is not the same as A->sourceSite, when yes, return null;
C) if A is falsy and B is truthy, then return null due to second condition group in nested IF
D) both falsy, do nothing (do not even return null)

translated to

```
A) if ($redirect->getSourceSite() && !$sourceSite){
    return null;
}
```

```
B) if ($sourceSite && $redirect->getSourceSite()) {
    if ($sourceSite->getId() !== $redirect->getSourceSite()) {
        return null;
    }
}
```
```
C) if (!$redirect->getSourceSite() && $sourceSite) {
    return null;
}
```


A and C are in a XOR situation 🤯 

which can be shortend in something like

```
if (($sourceSite xor $redirect->getSourceSite())){
    return null;
}else{
    if ($sourceSite && $sourceSite->getId() !== $redirect->getSourceSite()) {
        return null;
    }
}
```

Reply:
```php
if (($sourceSite xor $redirect->getSourceSite())){
    return null;
}
```
This can't be because if we have a source site for current request (this is $sourceSite) and the redirect does not have a source site (so it does not care), we should not return null.

Reply:
Ah, if that's the case, then the `C)` above was faulty

Reply:
I think 
 ```
A) if ($redirect->getSourceSite() && !$sourceSite){
    return null;
}
```
was missing in my simplification.

What do you think about
```php
if ($sourceSite && $sourceSite->getId() !== $redirect->getSourceSite()) {
    return null;
}

if (!$sourceSite && $redirect->getSourceSite()) {
    return null;
}
```
?

Reply:
Mhhh, not sure, feeling something is not right..  
Ideally, It looks like something we could set the expectation in a codeception test case and check how it behaves on each potential situation, would you be able to set something up for it?

Reply:
@kingjia90 I have added a test but it does not get executed. Can you guide me what to do to run the test?